The same three packages also failed on Darwin.

---

## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 453209`
Commit: `b5012be0584634f0823c764a95a1e5150b1e895e`

---
### `aarch64-darwin`
<details>
  <summary>:x: 3 packages failed to build:</summary>
  <ul>
    <li>ameba</li>
    <li>blahaj</li>
    <li>invidious</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 17 packages built:</summary>
  <ul>
    <li>ameba-ls</li>
    <li>amqpcat</li>
    <li>coverage-reporter</li>
    <li>crystal</li>
    <li>crystal.bin</li>
    <li>crystal.lib</li>
    <li>crystal2nix</li>
    <li>crystalline</li>
    <li>gi-crystal</li>
    <li>icr</li>
    <li>kakoune-cr</li>
    <li>lucky-cli</li>
    <li>mint</li>
    <li>oq</li>
    <li>shards</li>
    <li>thicket</li>
    <li>tmuxPlugins.fingers</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>ameba</summary>
<pre>       Expected:   &quot;\n&quot; +
       &quot;\n&quot; +
       &quot;\e[36m:1:5\e[39m\n&quot; +
       &quot;\e[34m[C] Ameba/DummyRule: DummyRuleError\e[39m\n&quot; +
       &quot;&gt; \e[97ma = 22\e[39m\n&quot; +
       &quot;      \e[33m^\e[39m\n&quot; +
       &quot;\n&quot; +
       &quot;Finished in 274 microseconds\n&quot; +
       &quot;\e[31m1 inspected, 1 failure\e[39m\n&quot;
       to include: &quot;      \e[33m^\e[0m&quot;

     # spec/ameba/formatter/dot_formatter_spec.cr:64

Finished in 790.03 milliseconds
1290 examples, 1 failures, 0 errors, 5 pending

Failed examples:

crystal spec spec/ameba/formatter/dot_formatter_spec.cr:53 # Ameba::Formatter::DotFormatter #finished when issues found writes affected code by default
make: *** [Makefile:52: spec] Error 1</pre>
</details>
<details>
<summary>blahaj</summary>
<pre>       &quot;\e[38;2;255;255;255;48;2;17;140;112m               ..  .............,*%%%%#%((((/                                   \e[39;49m\n&quot; +
       &quot;\e[38;2;0;0;0;48;2;45;205;169m                       **,,,****//*(##((###(#(((                                \e[39;49m\n&quot; +
       &quot;\e[38;2;0;0;0;48;2;151;231;192m                                        &amp;#(#/#((((((((#                         \e[39;49m\n&quot;

     # spec/colorizer_spec.cr:81

Finished in 8.09 milliseconds
16 examples, 7 failures, 0 errors, 0 pending

Failed examples:

crystal spec spec/colorizer_spec.cr:4 # Blahaj::Colorizer outputs a trans flag
crystal spec spec/colorizer_spec.cr:36 # Blahaj::Colorizer reads from ARGF and outputs a lesbian cowsay
crystal spec spec/colorizer_spec.cr:25 # Blahaj::Colorizer outputs an agender shark
crystal spec spec/colorizer_spec.cr:48 # Blahaj::Colorizer outputs an ace shark with each character colorized individually
crystal spec spec/colorizer_spec.cr:14 # Blahaj::Colorizer outputs a trans flag with 3x multiplier
crystal spec spec/colorizer_spec.cr:60 # Blahaj::Colorizer outputs an aro shark with each word colorized individually
crystal spec spec/colorizer_spec.cr:72 # Blahaj::Colorizer outputs a gay shark with its background colorized
Randomized with seed: 6780
make: *** [Makefile:24: test] Error 1</pre>
</details>
<details>
<summary>invidious</summary>
<pre>[3/13] Semantic (new)                    
[3/13] Semantic (new)                    
[4/13] Semantic (type declarations)      
[4/13] Semantic (type declarations)      
[5/13] Semantic (abstract def check)     
[5/13] Semantic (abstract def check)     
[6/13] Semantic (restrictions augmenter) 
[6/13] Semantic (restrictions augmenter) 
[7/13] Semantic (ivars initializers)     
[7/13] Semantic (ivars initializers)     
[8/13] Semantic (cvars initializers)     
[8/13] Semantic (cvars initializers)     
[9/13] Semantic (main)                   
Showing last frame. Use --error-trace for full trace.

In /nix/store/h08240caklfwx350kk2sh9g6slhpgvgz-crystal-1.18.1-lib/crystal/io.cr:933:92

 933 | def read_bytes(type : T.class, format : IO::ByteFormat = IO::ByteFormat::SystemEndian) : T forall T
                                                                                                ^
Error: method IO::Memory#read_bytes must return Protodec::VarLong but it is returning Int64</pre>
</details>
</details>

--- a/pkgs/by-name/am/ameba/package.nix
+++ b/pkgs/by-name/am/ameba/package.nix
@@ -1,10 +1,12 @@
 {
   lib,
   fetchFromGitHub,
-  crystal,
+  crystal_1_17,
   coreutils,
 }:
-
+let
+  crystal = crystal_1_17;
+in
 crystal.buildCrystalPackage rec {
   pname = "ameba";
   version = "1.6.4";

--- a/pkgs/by-name/bl/blahaj/package.nix
+++ b/pkgs/by-name/bl/blahaj/package.nix
@@ -1,13 +1,16 @@
 {
   lib,
   stdenv,
-  crystal,
+  crystal_1_17,
   fetchFromGitHub,
   # https://crystal-lang.org/2019/09/06/parallelism-in-crystal/
   multithreading ? true,
   static ? stdenv.hostPlatform.isStatic,
 }:
 
+let
+  crystal = crystal_1_17;
+in
 crystal.buildCrystalPackage rec {
   pname = "blahaj";
   version = "2.2.0";

--- a/pkgs/development/compilers/crystal/default.nix
+++ b/pkgs/development/compilers/crystal/default.nix
@@ -326,8 +326,8 @@ rec {
   };
 
   crystal_1_18 = generic {
-    version = "1.18.1";
-    sha256 = "sha256-6bJnonQyPtWIl1ex1tw3QercDl4ZQcvuVNYiWzMpGZ0=";
+    version = "1.18.2";
+    sha256 = "sha256-bwKs9bwD1WfS95DSxVY5AjT5Q61jOsfAH897tmiurng=";
     binary = binaryCrystal_1_10;
     llvmPackages = llvmPackages_21;
     doCheck = false;

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -4524,6 +4524,7 @@ with pkgs;
     crystal_1_15
     crystal_1_16
     crystal_1_17
+    crystal_1_18
     crystal
     ;
 

